repo/repo.go

```diff
@@ -107,15 +107,34 @@ func ImportRepoRules(filename string, repoCache *RemoteCache) ([]*rule.Rule, err
 	return rules, nil
 }
 
+// GetRulesToPrune returns a list of empty repo rules that can be pruned from the repo.
+// The list is determined by finding repo rules defined in the repository that no longer
+// have an equivalent repo defined in the Gopkg.lock/go.mod file.
+func GetRulesToPrune(rules []*rule.Rule, repos []Repo) []*rule.Rule {
```

Comment:
I think this logic could be integrated into `MergeRules`. Maybe pass a `pruneRules bool` parameter instead of `emptyRules`? It seems like that function has all the information it needs to build this list for each file.